### ISSUE TITLE
bind_device_by_index fails on IPv6

### ISSUE THREAD

**Issue by bryanlarsen** [rust-lang/socket2#413]
Calling `bind_device_by_index` results in errno 22 if the socket is bound to an IPV6 address.

I'm using this code instead:

```
    unsafe { libc::setsockopt(socket.as_raw_fd(), libc::IPPROTO_IPV6, libc::IPV6_BOUND_IF, &index as *const u32 as *const libc::c_void, std::mem::size_of::<u32>() as u32) } ;
```

`device_index` has the same problem.

P.S.  if somebody knows a better way of specifying a link-local scope ID when binding to an IPV6 socket, that would help me although the bug I opened is still valid.

**Comment by bryanlarsen** [comment]
Of course as soon as I hit submit, I find out the obvious answer to my P.S.: SocketAddrV6 has a scope_id field.

**Comment by Thomasdezeeuw** [comment]
I suppose this should have been IPv4 and IPv6 specific methods.

@pinkisemils you contributed the code, any thoughts?

**Comment by pinkisemils** [comment]
Yes, there should be an analogue IPv6 functions that would use `IPV6_BOUND_IF` and `IPPROTO_IPV6`. I'll make a small PR.

**Comment by Thomasdezeeuw** [comment]
We should probably add `_v4` and `_v6` methods, deprecating the no-suffix method.

**Comment by pinkisemils** [comment]
Yes. I dropped the ball on this due to my immune system dropping the ball on this. I'll add a small PR tomorrow.
